pkg/autogen/v2/autogen.go

```diff
@@ -301,3 +288,30 @@ func (a *ImplAutogenV2) ExtractPodSpec(resource unstructured.Unstructured) (*uns
 
 	return &unstructured.Unstructured{Object: podSpec}, nil
 }
+
+type ExtractPodFunc func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error)
+
+func getPodExtractor() ExtractPodFunc {
+	return func(resource *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		return extractPodSpec(*resource)
```

Comment:
don't dereference the pointer without a `nil` check. `extractPodSpec` should accept the pointer and do the nil check there